### PR TITLE
refactor: tx page react-query and typescript

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

- Use react-query instead of redux.
- Update Transaction component to typescript
- Remove ~200 lines of code

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### TypeScript/Hooks Update

- [x] Updated files to TypeScript

### CODE REVIEW

File: src/containers/Header/Banner.jsx [hunk 1]
```diff
@@ -22,7 +22,6 @@ Banner.propTypes = {
 export default connect((state) => {
   const messages = [
     ['ledgerError', state.ledger.error],
-    ['transactionError', state.transaction.error],
```
[the remark this answers is not in the document]
Comment (ckniffen) [reply src/containers/Header/Banner.jsx:25]: Correct.  The previous only showed the same message everytime and would not always disappear.

File: src/containers/test/utils.ts
Comment (ckniffen) [inline src/containers/test/utils.ts:1]: Will have a future PR to leverage this in other tests.

**Review by justinr1234 — APPROVED** [review]

File: src/containers/Transactions/index.tsx [hunk 2]
```diff
@@ -85,17 +108,18 @@ const Transaction = (props) => {
   }
 
   function renderTransaction() {
-    const { language, width } = props
     let body
 
+    if (!data) return <></>
```
Comment (khancode) [inline src/containers/Transactions/index.tsx:113]: nit: move above `let body`

**Review by khancode — APPROVED** [review]
LGTM with one nit comment

**Pull request merged**